rosbag2_storage_mcap/src/mcap_storage.cpp

```diff
@@ -815,8 +822,16 @@ void MCAPStorage::create_topic(const rosbag2_storage::TopicMetadata & topic,
     channel.topic = topic.name;
     channel.messageEncoding = topic_info.topic_metadata.serialization_format;
     channel.schemaId = schema_id;
-    channel.metadata.emplace("offered_qos_profiles",
-                             topic_info.topic_metadata.offered_qos_profiles);
+
+    std::vector<rosbag2_storage::Rosbag2QoS> to_encode;
```

Comment:
Here, a conversion from `std::vector<rclcpp::QoS>` to serialized YAML (`std::string`) is needed.